File: librarys/lib-logreport/app/src/main/java/com/doctorwork/android/logreport/MainActivity.java
package com.doctorwork.android.logreport;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void reportHttp(View view) {
        startActivity(new Intent(this, HttpActivity.class));
    }

    public void reportBusiness(View view) {
        startActivity(new Intent(this, BusinessActivity.class));
    }

    public void reportImg(View view) {
        startActivity(new Intent(this, ImgActivity.class));
    }

    public void reportCustom(View view) {
        startActivity(new Intent(this, CustomActivity.class));
    }

    public void reportPoint(View view) {
        startActivity(new Intent(this, PointActivity.class));
    }


}

File: librarys/lib-pay/app/src/main/java/com/medlinker/payapp/MainActivity.kt
package com.medlinker.payapp

import android.os.Bundle
import android.text.TextUtils
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.medlinker.lib.pay.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        PayAPI.getInstance().isDebugMode=true
    }

    fun testWechatPay(view: View) {
        var req = WxPayReq.Builder()
            .with(this)
            .setPayInfo("111")
            .create()
            .setOnWechatPayCallback(object : WxPayer.IPayCallback {
                override fun onFailure(errCode: Int, errStr: String?) {
                    toast(errStr)
                }

                override fun onSuccess(errCode: Int, errStr: String?) {
                    toast(errStr)
                }

            })
        PayAPI.getInstance().sendPayRequest(req);
    }

    fun testAliPay(view: View) {
        var req = AliPayReq.Builder()
            .with(this)
            .setPayInfo("111")
            .create()
            .setOnAliPayCallback(object : AliPayer.IPayCallBack {
                override fun onFailure(errStr: String?, resultStatus: String?) {
                    toast(errStr)
                }

                override fun onPaySuccess(
                    resultInfo: String?,
                    memo: String?,
                    resultStatus: String?
                ) {
                    toast(resultInfo)
                }

                override fun onPayIndeterminate(
                    resultInfo: String?,
                    memo: String?,
                    resultStatus: String?
                ) {
                    toast(resultInfo)
                }

            })
        PayAPI.getInstance().sendPayRequest(req)
    }

    fun toast(str: String?) {
        if (!TextUtils.isEmpty(str))
            Toast.makeText(MainActivity@ this, str, Toast.LENGTH_LONG).show()
    }

}